search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/prompt/PromptUtil.java

```diff
@@ -18,57 +18,144 @@
 package org.opensearch.searchpipelines.questionanswering.generative.prompt;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
 import lombok.AccessLevel;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
 import org.apache.commons.text.StringEscapeUtils;
+import org.opensearch.ml.common.conversation.Interaction;
 
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Locale;
 
 /**
+ * A utility class for producing prompts for LLMs.
+ *
  * TODO Should prompt engineering llm-specific?
  *
  */
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class PromptUtil {
 
+    public static final String DEFAULT_CHAT_COMPLETION_PROMPT_TEMPLATE =
```

Comment:
I remember you said you did some experiments on this prompt template to get the better result from LLM. I'm just curious to know more about that experiments. If you can share any doc/reference, that'll be helpful.

Reply:
Ack.  @HenryL27 can provide more details on this (but perhaps outside this PR?)

Reply:
> 

Yeah, sure. That's not going to block this PR.